### ISSUE TITLE
LeNet Lab Solution - Missing variable initialization

### ISSUE THREAD

**Issue by sahiljuneja** [udacity/sdc-issue-reports#295]
The code at the end of the solution -
```
with tf.Session() as sess:
    loader = tf.train.import_meta_graph('lenet.meta')
    loader.restore(sess, tf.train.latest_checkpoint('./'))

    test_accuracy = evaluate(X_test, y_test)
    print("Test Accuracy = {:.3f}".format(test_accuracy)) 
```

Should be updated to 

```
with tf.Session() as sess:
    sess.run(tf.global_variables_initializer())

    loader = tf.train.import_meta_graph('lenet.meta')
    loader.restore(sess, tf.train.latest_checkpoint('./'))

    test_accuracy = evaluate(X_test, y_test)
    print("Test Accuracy = {:.3f}".format(test_accuracy))
```

It was missing the variable initialization and throwing an error for those who tried to run it.

**Comment by sahiljuneja** [comment]
Okay, some other student said it works for them with or without initializing variables again. 
And the documentation seems to indicate it might not be required either. 

Should look into this more before making any changes I think just to be sure what could be the source of the discrepancy. I will also try to understand this more and update accordingly.

cc - @domluna, @dsilver829

**Comment by awbrown90** [comment]
I added a pull request for a simple change that gave me good results, where before I was having problems running the global variables init. I experimented with tensorflow v12 that is suppose to support global variable init but was then running into some other complications with import data_input. Using tf v11 along with the legacy run init all variables ran well.

**Comment by awbrown90** [comment]
Today I looked at the notebook while running on AWS with carnd-term1 environment and experienced no problems running through the entire notebook. While using tf init all variables  ran as well it is depreciated and will be removed after 2017-3-02 so probably best to keep using global init variable actually.  Once again no problems running the final code block without global init if the previous code block was already executed with global init. If you did use global init at the end you would actually reset all your variables so you don't want to do that. It looks like there have already been changes to the test accuracy block to correct what might have been the issue anyway.

**Comment by awbrown90** [comment]
Hi @sahiljuneja I think it's ok to go ahead and close this issue unless you wanted to try something else. I tested it on my end and it works, just need to execute sess.run(tf.global_variables_initializer()) on the previous code cell and it saves over to the test accuracy cell.
